File: project/1.16_多次元配列/java/java_ex_1_16_1/src/java_ex_1_16_1/Ex_1_16_1.java
package java_ex_1_16_1;

public class Ex_1_16_1 {

	public static void main(String[] args) {
		// Javaの多次元配列は、「配列の配列」となるので、↓が可能。
//		float[][] dataArr = {
//			{ -56.7f },
//			{ 500.1f, 70.70f },
//			{ 100.9f, 0.5f, 20.20f },	// 最後にカンマ入れてもOK
//		};

		// こんな領域確保もできるみたい。
		float[][] dataArr;
		dataArr = new float[][]{{-56.7f},{500.1f,70.70f},{100.9f,0.5f,20.20f}};		

		System.out.println( dataArr[0][0] );
//		System.out.println( dataArr[0][1] );	// 領域は割当たっていない
		System.out.println( dataArr[1][0] );
		System.out.println( dataArr[1][1] );
		System.out.println( dataArr[2][0] );
		System.out.println( dataArr[2][1] );
		System.out.println( dataArr[2][2] );

	}

}

File: project/1.16_多次元配列/py/py_ex_1_16_1/src/ex_1_16_1.py
'''
Created on 2023/03/12

@author: Keroichi.T
'''

def main():
    # 多次元配列は、「配列の配列」
    dataArr = [
        [ -56.7,],
        [ 500.1, 70.70],
        [ 100.9, 0.5, 20.20],
    ]
    
    # 指定した要素(オブジェクト)ごと抽出できる
    print( dataArr[0])
    print( dataArr[0][0])
    print( dataArr[1] )
    print( dataArr[1][0])
    print( dataArr[1][1])
    print( dataArr[2] )
    print( dataArr[2][0])
    print( dataArr[2][1])
    print( dataArr[2][2])
    
    dataArr = list( ["Python", "Java"] )
    print( dataArr )
    print( dataArr[0] )
    print( dataArr[0][2] )

if __name__ == '__main__':
    main()